packages/palette/src/elements/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,200 @@
+import React, {
+  Children,
+  createRef,
+  isValidElement,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react"
+import styled from "styled-components"
+import { useCursor } from "use-cursor"
+import { color, space } from "../../helpers"
+import { ChevronIcon } from "../../svgs"
+import { Box, BoxProps } from "../Box"
+import { Clickable } from "../Clickable"
+import { Skip } from "../Skip"
+import { VisuallyHidden } from "../VisuallyHidden"
+import { paginate } from "./utils"
+
+const Container = styled(Box)`
+  position: relative;
+  width: 100%;
+`
+
+const Viewport = styled(Box)`
+  width: 100%;
+  overflow: hidden;
+`
+
+const Rail = styled(Box)`
+  display: flex;
+  height: 100%;
+  transition: transform 500ms;
+  list-style: none;
+  margin: 0;
+  padding: 0;
+`
+
+const ARROW_WIDTH = space(4)
+
+const Arrow = styled(Clickable)`
```

Comment:
There are instances of the carousel (ViewingRooms, for example) that take custom left and right arrows; mind matching that bit of the existing API as well?

Reply:
I'm sort of split about this. Right now there are only two carousels that fall outside of a the consistent pattern that spec refers to: Viewing rooms and the main image view of the artwork page.

The artwork page one is an easy call: that has a very different UX and would benefit from a snowflake pulling from a few primitives that we might expose in Palette.

The viewing rooms one is similar here but still different enough: it anchors to center, pages a single cell at a time (vs an entire page), has overlaid arrows, a progress bar, and the mobile implementation varies (has the progress bar and arrows still, for instance). I suspect that one is also a snowflake.

My thinking is to implement it as a snowflake in force, but use a composition of pieces from palette rather than exposing a bunch of props, the existence of which might (will) encourage their usage.

So I can:
* expose the `paginate` function and add support a `center` alignment
* implement progress dots and a progress bar for the artwork images and viewing rooms, respectively
* implement a snapping/progress aware scrollable rail

Those pieces would let us build out both of those snowflakes in force without having to support a complicated API that would possibly lead to design inconsistencies.

In the meantime just to get off Flickity in most of the cases, we'll move Force's `Carousel` => `LegacyCarousel` and utilize it in both the viewing rooms and artwork pages.

We'll then build an adapter component in force and just ignore the following props: `width`, `height`, `options`, `arrowPosition`, `showArrows`, `renderLeftArrow`, `renderRightArrow`. Or just update the call sites to the new API, there's only 14 excluding the ❄️ .

Reply:
I like that strategy. It would be nice generally to push for more consistency and ensuring we have as simple of an api surface area as we can.

Reply:
This makes sense! Thanks for the thorough comment 👍